modules/aws/lb/main.tf

```diff
@@ -0,0 +1,272 @@
+  desync_mitigation_mode     = local.is_application ? var.desync_mitigation_mode : null
+  idle_timeout               = local.is_application ? var.idle_timeout : null
+  drop_invalid_header_fields = local.is_application ? var.drop_invalid_header_fields : null
+  enable_http2               = local.is_application ? var.enable_http2 : null
+  enable_waf_fail_open       = local.is_application ? var.enable_waf_fail_open : null
+
+  # Network Load Balancer specific settings
+  enable_cross_zone_load_balancing = local.is_network ? var.enable_cross_zone_load_balancing : null
+
+  dynamic "access_logs" {
+    for_each = var.access_logs != null ? [var.access_logs] : []
+    content {
+      bucket  = access_logs.value.bucket
+      prefix  = access_logs.value.prefix
+      enabled = access_logs.value.enabled
+    }
+  }
+
+  dynamic "subnet_mapping" {
+    for_each = var.subnet_mappings
+    content {
+      subnet_id            = subnet_mapping.value.subnet_id
+      allocation_id        = subnet_mapping.value.allocation_id
+      private_ipv4_address = subnet_mapping.value.private_ipv4_address
+      ipv6_address         = subnet_mapping.value.ipv6_address
+    }
+  }
+
+  tags = merge(
+    var.tags,
+    {
+      Name = var.name
+    },
+  )
+}
+
+resource "aws_lb_target_group" "target_group" {
+  for_each = var.target_groups
+
+  # Common settings
+  name                 = var.target_group_name
+  name_prefix          = var.target_group_name_prefix
+  port                 = var.target_group_port
+  protocol             = var.target_group_protocol
+  vpc_id               = var.target_group_vpc_id
+  target_type          = var.target_group_target_type
+  deregistration_delay = var.target_group_deregistration_delay
+
+  # Application Load Balancer specific settings
+  slow_start                    = local.is_application ? var.target_group_slow_start : null
+  load_balancing_algorithm_type = local.is_application ? var.target_group_load_balancing_algorithm_type : null
+
+  # Network Load Balancer specific settings
+  proxy_protocol_v2  = local.is_network ? var.target_group_proxy_protocol_v2 : null
+  preserve_client_ip = local.is_network ? var.target_group_preserve_client_ip : null
+
+  dynamic "health_check" {
+    for_each = each.value.health_check != null ? [each.value.health_check] : []
+    content {
+      enabled             = health_check.value.enabled
+      healthy_threshold   = health_check.value.healthy_threshold
+      interval            = health_check.value.interval
+      matcher             = health_check.value.matcher
+      path                = health_check.value.path
+      port                = health_check.value.port
+      protocol            = health_check.value.protocol
+      timeout             = health_check.value.timeout
+      unhealthy_threshold = health_check.value.unhealthy_threshold
+    }
+  }
+
+  dynamic "stickiness" {
+    for_each = each.value.stickiness != null ? [each.value.stickiness] : []
```

Comment:
Change to a map for consistency with the type:
```
for_each = each.value.stickiness != null ? each.value.stickiness : {}
```

Reply:
had to do set of object, map would not work for some reason, though I cannot remember exactly why.